pkg/kritis/review/review.go

```diff
@@ -52,12 +52,26 @@ func New(client metadata.Fetcher, c *Config) Reviewer {
 	}
 }
 
+// Flatten allowlist from a list of generic attestation policies
+func generateGapAllowlist(gaps []v1beta1.GenericAttestationPolicy) []string {
+	allowlist := []string{}
+	for _, gap := range gaps {
+		for _, pattern := range gap.Spec.AdmissionWhitelistPatterns {
+			allowlist = append(allowlist, pattern.NamePattern)
+		}
+	}
+	return allowlist
+}
+
 // ReviewGAP reviews images against generic attestation policies
 // Returns error if violations are found and handles them per violation strategy
 func (r Reviewer) ReviewGAP(images []string, gaps []v1beta1.GenericAttestationPolicy, pod *v1.Pod) error {
 	images = util.RemoveGloballyAllowedImages(images)
+
+	images = util.RemoveGapAllowedImages(images, generateGapAllowlist(gaps))
```

Comment:
It is odd to flatten allowlists from different policies into a single allowlist.  Allowlist should be handled separately for each policy.  On the other hand the outer loop below is by image, so not clear what to do.

We could invert the loops below (by gap first, by image then), and then allowlist handling can be per policy.  WDYT?

Reply:
The biggest reason in doing it this way now is now to have kritis api dependency in `util` -- which I don't have strong opinion on. Or we can have this function in `genericattestation.go`.

Reply:
I did not follow this.  Which dependency are you trying to avoid?

Reply:
Let's postpone this until ReviewGAP() is restructured as discussed (for loops inverted).  LGTM as is.

Reply:
Resolving this now since the loop inversion is only possible after #432 is addressed.